### PR TITLE
Merge created pipeline to develop

### PR DESCRIPTION
**Pull request by zombori**

### CODE REVIEW

File: backend-refactor/OutOfDomainDetection.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class OutOfDomainDetection:
```
Comment (qeterme) [inline backend-refactor/OutOfDomainDetection.py:1]: Shouldn't the different pipeline steps be inherited from a common class?
```suggestion
class OutOfDomainDetection(PipelineStep):
```

File: backend-refactor/Pipeline.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                 data: dict,
+                 out_of_domain_detection: OutOfDomainDetection,
+                 retriever: Retriever,
+                 retriever_agreg: RetrieverAgreg,
+                 reader: Reader):
+        self._data: dict = data
+        self._out_of_domain_detection: OutOfDomainDetection = out_of_domain_detection
+        self._retriever: Retriever = retriever
+        self._retriever_agreg: RetrieverAgreg = retriever_agreg
+        self._reader: Reader = reader
```
Comment (qeterme) [inline backend-refactor/Pipeline.py:58]: With the inheritance thing, we don't need different types of variables in the pipeline

File: backend-refactor/OutOfDomainDetection.py (hunk 1 above)
Comment (zombori) [reply backend-refactor/OutOfDomainDetection.py]: I created the common class which contains one abstract method

File: backend-refactor/Pipeline.py (hunk 2 above)
Comment (zombori) [reply backend-refactor/Pipeline.py]: replaced it with list of tuples which is a name, function pair

**Pull request merged**